app/models/like_entry.rb

```diff
@@ -1,4 +1,10 @@
 class LikeEntry < ApplicationRecord
-  belongs_to :author
   belongs_to :post
+  belongs_to :author, class_name: 'User'
+
+  after_create :update_post_likes_counter
+
+  def update_post_likes_counter
+    post.update(likes_counter: post.like_entries.count)
+  end
```

Comment:
:red_circle: [REQUIRED] The method you provided is not correct because it does not increment the likes_counter attribute. The update method takes a hash as its argument, and the key-value pair likes_counter: post.like_entries.count tells the update method to update the likes_counter attribute to the current number of likes the post has received.

However, this method will not increment the likes_counter attribute by 1 every time a new like is added. If the post already has 1 like, then this method will not change the value of the likes_counter attribute.

The better approach to increment the likes_counter attribute by 1 every time a new like is added is to use the increment! method. The increment! method will increment the value of the attribute by 1, and it will also save the changes to the database.

Here is the corrected code:

```
def update_post_likes_counter
  # Increment the likes_counter attribute by 1.
  post.increment!(:likes_counter)
end

```

This code will first find the Post object. Then, it will increment the likes_counter attribute by 1. Finally, it will save the changes to the database.

Reply:
@bienvenuushindi done